app/src/main/java/com/work/restaurant/view/adapter/ViewPagerAdapter.kt

```diff
@@ -3,26 +3,44 @@ package com.work.restaurant.view.adapter
 import androidx.fragment.app.Fragment
 import androidx.fragment.app.FragmentManager
 import androidx.fragment.app.FragmentPagerAdapter
+import com.work.restaurant.view.calendar.CalendarFragment
+import com.work.restaurant.view.diary.main.DiaryFragment
+import com.work.restaurant.view.home.main.HomeFragment
+import com.work.restaurant.view.mypage.main.MyPageFragment
+import com.work.restaurant.view.search.main.SearchFragment
 
 
 class ViewPagerAdapter(
     supportFragmentManager: FragmentManager,
-    private val fragmentMap: Map<String, Fragment>
+    private val tabList: List<String>
 ) :
     FragmentPagerAdapter(supportFragmentManager, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
 
-
     override fun getItem(position: Int): Fragment =
```

Comment:
```suggestion
    abstract override fun getItem(position: Int):  Fragment
```
이렇게 추상 클래스 및 추상 메소드로 변경한 후 caller에서 아래 when 문을 구현하면 재사용이 됩니다.

Reply:
79c6b4c 반영하였습니다.